bec_lib/tests/test_serializer.py

```diff
@@ -177,3 +177,27 @@ def test_legacy_serialize(data):
     # Can be removed in the future, cf issue #516
     res = deprecated_msgpack.loads(deprecated_msgpack.dumps(data)) == data
     assert all(res) if isinstance(data, np.ndarray) else res
+
+
+def test_serializer_registry_cache_resets():
+    """
+    Test that adding a new codec resets the cache.
+    """
+
+    class DummyType:
+        pass
+
+    class DummyCodec(BECCodec):
+        obj_type = DummyType
+
+        @staticmethod
+        def encode(obj):
+            return {"dummy": "data"}
+
+        @staticmethod
+        def decode(type_name: str, data: dict):
+            return DummyType()
+
+    assert msgpack.is_registered(DummyType) is False
+    msgpack.register_codec(DummyCodec)
+    assert msgpack.is_registered(DummyType) is True
```

Comment:
Do you prefer this syntax? I would have written:

```suggestion
    assert not msgpack.is_registered(DummyType)
    msgpack.register_codec(DummyCodec)
    assert msgpack.is_registered(DummyType)
```

because it is a bit superfluous to add another comparison when you already have a bool to assert, but it is a matter of taste what is more readable.